### PR TITLE
Fix crashes in pydiar

### PR DESCRIPTION
**Pull request by pajowu**
This bumps the pydiar version and simply catches all errors in the diarization code.

### CODE REVIEW

File: server/app/transcribe.py [hunk 1]
```diff
@@ -106,12 +107,18 @@ def process_audio(
...
+        try:
+            diarization_model = BinaryKeyDiarizationModel()
+            diarization_model.CLUSTERING_SELECTION_MAX_SPEAKERS = diarize_max_speakers
+            segments = diarization_model.diarize(
+                SAMPLE_RATE, np.array(audio.get_array_of_samples())
+            )
+            optimized_segments = optimize_segments(segments)
+        except:  # noqa: E722
+            traceback.print_exc()
+            optimized_segments = [
```
Comment (anuejn) [inline server/app/transcribe.py:119]: we probably want to display a warning if diarization failed. though we can do this in another pr

**Review by anuejn — APPROVED** [review]

File: server/app/transcribe.py (hunk 1 above)
Comment (pajowu) [reply server/app/transcribe.py:119]: I opened #273 for that
